test/node/connection-emit-actions.js

```diff
@@ -0,0 +1,221 @@
+      test.fail('must create socket and connect to server');
+    });
+    socket.on('connect', () => {
+      serverConnection.on('handshakeRequest',
+        (applicationName, authStrategy) => {
+          test.equal(applicationName, app.name,
+            'application name must match');
+          test.equal(authStrategy, 'login',
+            'auth strategy must be \'login\' with login/password');
+        }
+      );
+
+      const transport = new jstp.net.Transport(socket);
+      const connection = new jstp.Connection(transport, null, client);
+
+      connection.on('handshake', (error, ok) => {
+        test.assertNot(error, 'handshake must not return an error');
+        test.equal(ok, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+      });
+
+      connection.handshake(app.name, app.login, app.password, (error) => {
+        test.assertNot(error, 'handshake must not return an error');
+        test.equal(connection.username, app.login, 'username must match');
+        test.equal(connection.sessionId, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+        connection.close();
+      });
+    });
+  }
+);
+
+test.test('must emit event on call with no arguments and no return value',
+  (test) => {
+    test.plan(5);
+
+    const iface = 'calculator';
+    const methodName = 'doNothing';
+    const args = [];
+    addCallEventCheck(test, server.getClients()[0], iface, methodName, args);
+    connection.on('callback', (error, ok) => {
+      test.assertNot(error, 'callMethod must not return an error');
+      test.strictSame(ok, [], 'Ok contents must match');
+    });
+    connection.callMethod(iface, methodName, args);
+  }
+);
+
+test.test('must emit event on call with no arguments and return value',
+  (test) => {
+    test.plan(5);
+
+    const iface = 'calculator';
+    const methodName = 'answer';
+    const args = [];
+    addCallEventCheck(test, server.getClients()[0], iface, methodName, args);
+    connection.on('callback', (error, ok) => {
+      test.assertNot(error, 'callMethod must not return an error');
+      test.strictSame(ok, [42], 'Ok contents must match');
+    });
+    connection.callMethod(iface, methodName, args);
+  }
+);
+
+test.test('must emit event upon inspect packet', (test) => {
+  const expectedInterfaces = Object.keys(app.interfaces);
+  const expectedTests = expectedInterfaces.length;
+
+  test.plan(expectedTests);
+  server.getClients()[0].on('inspect', (interfaceName) => {
+    test.assert(expectedInterfaces.includes(interfaceName),
+      'inspect event interface must be one of expected');
+  });
+
+  expectedInterfaces.forEach((iface) => {
+    connection.inspectInterface(iface);
+  });
+});
+
+test.test('must emit packets in not "production" mode', (test) => {
+  // 4 packets from call below and 4 from 1 heartbeat
+  test.plan(8);
+
+  const clientSentPackets = [{}, { call: [1, 'calculator'], answer: [] }];
```

Comment:
@nechaido You meant heartbeat? I don't really care, I thought that we should test that it emits heartbeat packets.